interop/xds/server/server.go

```diff
@@ -78,8 +87,48 @@ func (s *testServiceImpl) EmptyCall(ctx context.Context, _ *testpb.Empty) (*test
 }
 
 func (s *testServiceImpl) UnaryCall(ctx context.Context, in *testpb.SimpleRequest) (*testpb.SimpleResponse, error) {
+	response := &testpb.SimpleResponse{ServerId: s.serverID}
+
+	for _, val := range getRPCBehaviorMetadata(ctx) {
+		if st := getStatusForRPCBehaviorMetadata(val); st != nil {
+			return response, status.Err(st.Code(), st.Message())
+		}
+	}
+
+	response.Hostname = s.hostname
 	grpc.SetHeader(ctx, metadata.Pairs("hostname", s.hostname))
-	return &testpb.SimpleResponse{ServerId: s.serverID, Hostname: s.hostname}, nil
+	return response, status.Err(codes.OK, "")
+}
+
+func getRPCBehaviorMetadata(ctx context.Context) []string {
+	md, ok := metadata.FromIncomingContext(ctx)
+	if !ok {
+		logger.Error("failed to receive metadata")
```

Comment:
Log strings don't need to follow the rules for error strings about not having them capitalized. So, this could instead be: `logger.Error("Failed to retrieve metadata from incoming RPC context")`